### ISSUE TITLE
3,000,000 HYDRO to Create a Predictions, Odds, Payouts and Probabilities Ethereum Smart Contract

### ISSUE THREAD

**Issue by mitdralla** [HydroBlockchain/hcdp#263]
## Project Details ##
3,000,000 HYDRO to create an Ethereum smart contract on top of [Hydro Snowflake](https://medium.com/hydrogen-api/examining-erc-1484-hydro-snowflake-cc3753a5ff52) that ties given predictions, odds, payouts, and probabilities to a Snowflake address and allows multiple addresses to setup a marketplace that stores. This utility smart contract will power Hydro dApps that use prediction markets and game theory dynamics. There will be Layer-3 dApps and Layer-4 APIs that hook into this utility smart contract function.

### Background: ###
* Prediction markets have existed for thousands of years across finance, politics, sports, and ther weather. 
* The market opportunity for prediction markets is tens of billions of dollars
* Existing prediction markets on Ethereum, such as Augur have been heavily criticized for their mechanics and costs, this stems from poor infrastructure and heavy centralization

### Features: ###
* **Create prediction** - define what the prediction is
* **Create payout if true** - amount of HYDRO to be released if TRUE
* **Create payout if false** - amount of HYDRO to be released if FALSE
* **Assign** - assign a prediction to a Snowflake ID or group of IDs
* **Probability** - define probability for a prediction
* **Odds** - define odds for a prediction, default is 1:1
* **End Date** - set the end date of the prediction. 
* **Escrow** - remove the set amount of HYDRO from the SnowflakeID wallet to an escrow contract on the payment schedule intervals, to insure there will be no default
* **Send payout** - distribute the interest HYDRO from the escrow on the Payment Schedule date
* **Collect fee** - ability to collect fee in HYDRO as an intermediary in a prediction market
* **Confirm Payment** - use the Hydro Raindrop smart contract to confirm payout
* **Dispute** - reate a flag for a disputed payout

## What is the development you want done? ##
Solidity code, in the form of an open-source smart contract, deployed to the Hydro Blockchain GitHub repository for public use.

## What are the deliverables? ##
Solidity code, in the form of an open-source smart contract, deployed to the Hydro Blockchain GitHub repository for public use.

## Timeframe? ##
We propose 3-4 weeks as the timeframe for completion of this task.

## Submission process? ##
First, **_comment on this issue to declare your willingness to complete the task_**. Next, fill out the [Off-Chain Smart Contract Claim Form](https://docs.google.com/forms/d/1Y_hISwogGEKQOq_toXanN7c8p5atcGEsRa9tC4ovDxk). We will evaluate candidates factoring in their qualifications, Github activity, and the order in which they expressed interest.

## Oversight Process ##
One of the DAs will be assigned to oversee the completion of this task once it’s awarded.

## Reward Structure ##
Once you have been selected, you will have **3-4 weeks** to complete the task. The time frame begins as soon as we award the task to the candidate by communication within this thread.
During this time we will be able to provide some support around questions about the workflow on Discord.

During this time the candidate needs to update the codebase regularly in a subrepo of our [official Github](https://github.com/HydroBlockchain).  This will be setup and conveyed to the candidate by the assigned DA.

If the task is not completed within the timeframe we reserve the right to offer an extension, or transfer the task to another candidate. If no extension is granted, then we reserve the right to continue on the codebase committed. The fractional disbursement for it, if any, would be given at our discretion.

Once the task is completed we will test the plug-in for efficiency and whether or not it correctly incorporates the Hydro workflow. If we approve the plug-in, we will then send you 3,000,000 Hydro.

**Comment by amiromayer** [comment]
I want this project!

**Comment by 34x4p08** [comment]
I can do this!

**Comment by Luiserebii** [comment]
I'm really interested in this!

**Comment by mitdralla** [comment]
This task is assigned to @amiromayer - we will reach out with next steps. 👍

**Comment by HarshRajat** [comment]
Hi @amiromayer, please respond if you would like to continue on this task or not. We can give you a time extension. I will be pulling this bounty if there is no answer within the next 48 hours.

**Comment by HarshRajat** [comment]
Hi @amiromayer, please respond if you would like to continue on this task or not. We can give you a time extension. I will be pulling this bounty if there is no answer within the next **24 hours**.

**Comment by HarshRajat** [comment]
@amiromayer I just wanted to let you know this HCDP bounty has officially been revoked. We hope you can be involved in the future.

Will be re-assigning this shortly.

**Comment by HarshRajat** [comment]
The task will be re-assigned to @Luiserebii if he is still interested in it.

**Comment by Luiserebii** [comment]
@HarshRajat Yes, I am still interested in taking this on! Would love to try and tackle this.

**Comment by HarshRajat** [comment]
Task has been assigned to @Luiserebii, we will reach out with next steps.
